File: Data Structures & Algorithms/search-2d-matrix/submission-0.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        return searchinMatrix(matrix, target, 0, matrix.length -1);
    }

    public boolean searchinMatrix(int[][] matrix, int target, int left, int right) {
        if (left > right) return false;
        int mid = left + (right - left) / 2;

        if (matrix[mid][0] > target) {
            return searchinMatrix(matrix, target, left, mid -1);
        }
        if (matrix[mid][matrix[mid].length -1] < target) {
            return searchinMatrix(matrix, target, mid +1, right);
        }
        return binarySearch(matrix[mid], target, 0, matrix[mid].length -1);
    }

    public boolean binarySearch(int[] arr, int target, int left, int right) {
        if (left > right) return false;
        int mid = left + (right - left) / 2;
        
        if (arr[mid] == target) return true;

        if (arr[mid] > target) {
            return binarySearch(arr, target, left, mid -1);
        }
        return binarySearch(arr, target, mid + 1, right);
    }
}

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def levelOrderRec(self, node: Optional[TreeNode], level: int, arr: [int]):
        if not node:
            return
        
        if len(arr) <= level:
            arr.append([])

        arr[level].append(node.val)

        self.levelOrderRec(node.left, level + 1, arr)
        self.levelOrderRec(node.right, level + 1, arr)

    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        array = []
        self.levelOrderRec(root, 0, array)
        return array
        